src/Data/Quaternion/Rotation.purs

```diff
@@ -107,3 +112,24 @@ act (Rotation p) v =
   unsafePartial $ case Vec.toArray v of
     [x, y, z] ->
       vectorPart (Quaternion zero x y z `conjugateBy` p)
+
+normalize :: Rotation Number -> Rotation Number
```

Comment:
If a Quaternion is used as the state of a numerical integrator (kinematics) its magnitude can drift with time. Also, there is an alternative to slerp (sometimes just "lerp") that avoids some transcendental calls with linear interpolation followed by normalization. This can be quite a bit faster, with bounded error.